Singularity/Singularity/Platform/PlatformBlank.cs

```diff
@@ -63,36 +68,38 @@ public void UnAssignUnits(GeneralUnit unit)
         }
 
         /// <summary>
-        /// Get the special actions you can perform on this platform.
+        /// Get the special IPlatformActions you can perform on this platform.
         /// </summary>
-        /// <returns> an array with the available actions.</returns>
-        public Action[] GetActions()
+        /// <returns> an array with the available IPlatformActions.</returns>
+        public IPlatformAction[] GetIPlatformActions()
         {
-            return mActions;
+            return mIPlatformActions;
         }
 
         /// <summary>
-        /// Perform the given action on the platform.
+        /// Perform the given PlatformAction on the platform.
         /// </summary>
-        /// <param name="action"> The action to be performed </param>
+        /// <param name="platformAction"> The IPlatformAction to be performed </param>
         /// <returns> true if it was succesfull</returns>
-        public bool DoAction(Action action)
+        public bool DoIPlatformAction(IPlatformAction platformAction)
         {
-            //This return is normally an if, I just had to do it this way because resharper would cry otherwise. As soon as doBlueprintBuild is implemented we can change this.
-            return (action == Action.BlueprintBuild);
-            //{
-                //doBlueprintBuild
-                //return true;
-            //}
+            // FIXME might need to give the ID instead
```

Comment:
What do you mean by that?

Reply:
That was the thing about id's from PlatformActions I think